### PR TITLE
Badges react

### LINKED ISSUE #1034: JSX: badges

### PR DESCRIPTION
**Pull request by anna-wro**
screen shot 2017-07-11 at 15 59 04

Closes #1034

### CODE REVIEW

File: src/components/badges/badge.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+test('size', () => {
+  const badge = shallow(
+    <Badge size={SIZE.SMALL}>Test</Badge>
+  );
+
+  expect(badge.hasClass('sg-badge--small')).toEqual(true);
+});
+
+test('default font size', () => {
+  const badge = mount(
```
Comment (matikrk) [inline src/components/badges/badge.spec.jsx]: You can test if using Text component in render method. And then check if you use correct props, not directly checking text css class (it's part of Text.spec)

File: src/components/badges/badge.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    <Badge>Test</Badge>
+  );
+  const text = badge.find(Text);
+
+  expect(text.hasClass('sg-text--xsmall')).toEqual(true);
+});
+
+test('larger font size', () => {
```
Comment (matikrk) [inline src/components/badges/badge.spec.jsx]: what behaviour have small font?

File: src/components/badges/badge.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    <Badge size={SIZE.LARGE}>Test</Badge>
+  );
+  const text = badge.find(Text);
+
+  expect(text.hasClass('sg-text--xsmall')).toEqual(false);
+});
+
+test('color', () => {
+  const badge = mount(
+    <Badge color={COLOR.MINT_SECONDARY}>Test</Badge>
```
Comment (matikrk) [inline src/components/badges/badge.spec.jsx]: shallow copy and check if passing props correctly - single responsibility

File: src/components/badges/badge.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+test('rounded', () => {
+  const badge = shallow(
+    <Badge rounded={true}>Test</Badge>
+  );
+
+  expect(badge.hasClass('sg-badge--rounded')).toEqual(true);
+});
+
+test('error when no child', () => {
```
Comment (matikrk) [inline src/components/badges/badge.spec.jsx]: pls add fail test when passing some div instead of string to show badge behaviour

Comment (anna-wro) [inline src/components/badges/badge.spec.jsx]: Done! Thanks for your suggestions:)

File: src/components/badges/Badge.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import Text, {SIZE as TEXT_SIZE, WEIGHT as TEXT_WEIGHT} from '../text/Text';
+
+const SIZE = {
+  NORMAL: 'normal',
+  SMALL: 'small',
+  LARGE: 'large'
+};
+
+const COLOR = {
+  NORMAL: {BADGE: 'light', TEXT: 'default'},
```
Comment (kdzwinel) [inline src/components/badges/Badge.jsx]: text colors are defined in Text component, you should not use strings here but the constants. Do sth like this instead:

```javascript
import Text, {COLOR as TEXT_COLOR} from '../text/Text';
//...
NORMAL: {BADGE: 'light', TEXT: TEXT_COLOR.DEFAULT},
```

Comment (matikrk) [inline src/components/badges/Badge.jsx]: good point 👍

Comment (anna-wro) [inline src/components/badges/Badge.jsx]: You're right, I've changed it 👍

**Pull request merged**